source/OctoVersion.Core/OctoVersionInfo.cs

```diff
@@ -1,44 +1,50 @@
 ﻿using System;
 using System.Text.RegularExpressions;
+using OctoVersion.Core.VersionTemplates;
 
 namespace OctoVersion.Core
 {
     public class OctoVersionInfo : SemanticVersion
     {
-        //https://semver.org/spec/v2.0.0.html#spec-item-9
-        static readonly Regex InvalidPreReleaseCharacters = new Regex("[^0-9A-Za-z-\\.]", RegexOptions.Compiled);
+        readonly VersionParser _versionParser;
```

Comment:
This class is supposed to be the "answer" class. Why does it need to take a dependency on a service?